src/classes/ADDR_Account_TDTM.cls

```diff
@@ -48,123 +51,122 @@ public class ADDR_Account_TDTM extends TDTM_Runnable {
     * @param dmlWrapper to hold the Addresses that need creating
     * @return void
     ********************************************************************************************************/
-    private void createAddrFromAcc(list<Account> listAcc, DmlWrapper dmlWrapper) {
-        list<Address__c> listAddr = new list<Address__c>();
-        for (Account acc : listAcc) {
+    private void createAddrFromAcc(List<Account> ListAcc, DmlWrapper dmlWrapper) {
+        List<Address__c> ListAddr = new List<Address__c>();
+        for (Account acc : ListAcc) {
             Address__c addr = new Address__c();
             addr.Parent_Account__c = acc.Id;
             addr.Default_Address__c = true;
             addr.Latest_Start_Date__c = system.today();
             addr.Latest_End_Date__c = null;
-            ADDR_Addresses_TDTM.copyAddressStdSObjAddr(acc, 'Billing', addr, null);            
-            listAddr.add(addr);         
+            ADDR_Addresses_UTIL.copyAddressStdSObjAddr(acc, 'Billing', addr, null);            
+            ListAddr.add(addr);
+            UTIL_Debug.debug('****New address record: ' + JSON.serializePretty(addr));
         }
-        // since coming from an account address, there is no Address Type, so exclude it from the match testing.
-        ADDR_Addresses_TDTM.createNonDuplicateAddresses(listAddr, dmlWrapper, false);
+        //since coming from an account address, there is no Address Type, so exclude it from the match testing.
+        List<Address__c> newAddrs = ADDR_Addresses_UTIL.createNonDuplicateAddresses(ListAddr, dmlWrapper, false);
+        
+        //Match each Account with each new address
+        List<ID> accIDsToUpdate = new List<ID>();
+        Map<ID, Address__c> parentAccIdToAddrMap = new Map<ID, Address__c>();
+        for(Address__c addr : newAddrs) {
+            if(addr.Parent_Account__c != null) {
+	           accIDsToUpdate.add(addr.Parent_Account__c);
+	           parentAccIdToAddrMap.put(addr.Parent_Account__c, addr);  
+            }
+        }
+        
+        //We need to re-query before we are in the "after" part of the trigger. We cannot change it to before because
+        //we need the Account ID to have a value when populating the addr.Parent_Account__c field above.
+        List<Account> parentAccs = [select is_Address_Override__c, Current_Address__c from Account where ID in :accIDsToUpdate];
+        List<Account> accsToUpdate = new List<Account>();
+        for(Account acc : parentAccs) {
+            Address__c childAddr = parentAccIdToAddrMap.get(acc.Id);
+            if(childAddr != null && !acc.is_Address_Override__c) {
+                acc.Current_Address__c = childAddr.Id;
+                accsToUpdate.add(acc);
+            }
+        }
+        update accsToUpdate;
+    }
+    
+    private void addrInfoDeleted(List<Account> ListAccsAddrInfoCleared, DmlWrapper dmlWrapper) {
+        List<ID> oldCurrentAddrIDs = new List<ID>();
```

Comment:
Variables should use the `lowerCamelCase` naming convention. Please rename accordingly.